### PR TITLE
[plugin] withProgress should start task immediately

### PR DESCRIPTION
**Pull request by AlexTugarev**
#### What it does
Fixes delayed task execution in `withProgress`. The resolution of the promise returned by the task may in some cases depend on the task being executed asap, e.g. the python extension expects that the task is running immediately in order to create a deferred promise, which if delayed breaks the removal of progress. 

In case of python extension it led to the never ending `Refreshing Python Interpreters` progress.

Screen Shot 2019-09-06 at 09 14 29

#### How to test
1. install vscode extension for python and open a .py file
2. verify that `Refreshing Python Interpreters` will be hidden when done

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/plugin/notification.ts [hunk 1]
```diff
@@ -31,15 +32,18 @@ export class NotificationExtImpl implements NotificationExt {
         options: ProgressOptions,
         task: (progress: Progress<{ message?: string; increment?: number }>, token: CancellationToken) => PromiseLike<R>
     ): Promise<R> {
+        const id = new Deferred<string>();
+        const tokenSource = new CancellationTokenSource();
+        const progress = task({ report: async item => this.proxy.$updateProgress(await id.promise, item)}, tokenSource.token);
```
Comment (akosyakov) [inline packages/plugin-ext/src/plugin/notification.ts]: Does it guarantee that the last item reported?

Comment (AlexTugarev) [reply packages/plugin-ext/src/plugin/notification.ts]: @akosyakov 'you mean making the report function async could lead to out-of-order `$updateProgress` calls?

Comment (AlexTugarev) [reply packages/plugin-ext/src/plugin/notification.ts]: it's the same promise they would be waiting for, shouldn't the event loop process them in order?

Comment (akosyakov) [reply packages/plugin-ext/src/plugin/notification.ts]: yes, i think it should

**Pull request merged**